include/operations/blas1_trees.hpp

```diff
@@ -248,6 +248,25 @@ struct BinaryOp {
   }
 };
 
+/*! TupleOp.
+ * @brief Implements a Tuple Operation (map (\x -> [i, x]) vector).
+ */
+template <typename RHS>
+struct TupleOp {
+  using value_type = IndVal<typename RHS::value_type>;
+  RHS r;
+
+  TupleOp(RHS &_r) : r(_r) {}
+
+  size_t getSize() { return r.getSize(); }
+
+  value_type eval(size_t i) { return value_type(i, r.eval(i)); }
+
+  value_type eval(cl::sycl::nd_item<1> ndItem) {
+    return eval(ndItem.get_global(0));
+  }
+};
+
```

Comment:
At least for the time being, it is only being used for amax/amin, so it belongs to blas1_tree header.